server/container/guice/postgres-common/src/main/java/org/apache/james/modules/data/PostgresCommonModule.java

```diff
@@ -76,14 +76,14 @@ PostgresConfiguration provideConfiguration(PropertiesProvider propertiesProvider
     @Singleton
     JamesPostgresConnectionFactory provideJamesPostgresConnectionFactory(PostgresConfiguration postgresConfiguration,
                                                                          ConnectionFactory connectionFactory,
-                                                                         @Named(JamesPostgresConnectionFactory.NON_RLS_INJECT) JamesPostgresConnectionFactory singlePostgresConnectionFactory) {
+                                                                         @Named(JamesPostgresConnectionFactory.NON_RLS_INJECT) JamesPostgresConnectionFactory jamesPostgresConnectionFactory) {
         if (postgresConfiguration.rowLevelSecurityEnabled()) {
             LOGGER.info("PostgreSQL row level security enabled");
-            LOGGER.info("Implementation for PostgreSQL connection factory: {}", DomainImplPostgresConnectionFactory.class.getName());
-            return new DomainImplPostgresConnectionFactory(connectionFactory);
+            LOGGER.info("Implementation for PostgreSQL connection factory: {}", PoolBackedPostgresConnectionFactory.class.getName());
+            return new PoolBackedPostgresConnectionFactory(true, connectionFactory);
         }
-        LOGGER.info("Implementation for PostgreSQL connection factory: {}", SinglePostgresConnectionFactory.class.getName());
-        return singlePostgresConnectionFactory;
+        LOGGER.info("Implementation for PostgreSQL connection factory: {}", PoolBackedPostgresConnectionFactory.class.getName());
+        return new PoolBackedPostgresConnectionFactory(false, connectionFactory);
```

Comment:
What that boolean is? (name the constant!)

Also the `if` could be flattened?

`return new PoolBackedPostgresConnectionFactory(postgresConfiguration.rowLevelSecurityEnabled(), connectionFactory);`

Reply:
As far as I can tell, comment is not handled.